sys/stdio_uart/stdio_uart.c

```diff
@@ -45,16 +47,28 @@ extern ethos_t ethos;
 #define ENABLE_DEBUG 0
 #include "debug.h"
 
-
+#ifdef MODULE_STDIO_UART_RX
 static char _rx_buf_mem[STDIO_UART_RX_BUFSIZE];
 isrpipe_t stdio_uart_isrpipe = ISRPIPE_INIT(_rx_buf_mem);
+#endif
 
 void stdio_init(void)
 {
+    uart_rx_cb_t cb;
+    void *arg;
+
+#ifdef MODULE_STDIO_UART_RX
+    cb = (uart_rx_cb_t) isrpipe_write_one;
+    arg = &stdio_uart_isrpipe;
+#else
+    cb = NULL;
+    arg = NULL;
+#endif
+
 #ifndef USE_ETHOS_FOR_STDIO
-    uart_init(STDIO_UART_DEV, STDIO_UART_BAUDRATE, (uart_rx_cb_t) isrpipe_write_one, &stdio_uart_isrpipe);
+    uart_init(STDIO_UART_DEV, STDIO_UART_BAUDRATE, cb, arg);
 #else
-    uart_init(ETHOS_UART, ETHOS_BAUDRATE, (uart_rx_cb_t) isrpipe_write_one, &stdio_uart_isrpipe);
+    uart_init(ETHOS_UART, ETHOS_BAUDRATE, cb, arg);
```

Comment:
I think we need to error here. ethos won't work without rx.

Reply:
Is my last commit enough to fix this?

Reply:
Perfect!